code/game/objects/effects/misc.dm

```diff
@@ -45,18 +45,106 @@
 	opacity = FALSE
 	icon_state = "speaker"
 	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
-	var/datum/looping_sound/alarm_loop/deltalarm
 
 /obj/effect/soundplayer/Initialize(mapload)
 	. = ..()
-	deltalarm = new(null, FALSE)
-	GLOB.ship_alarms += src
 	icon_state = ""
 
-/obj/effect/soundplayer/Destroy()
+/obj/effect/soundplayer/deltaplayer
+	var/datum/looping_sound/alarm_loop/deltaloop
+
+/obj/effect/soundplayer/deltaplayer/Initialize(mapload)
+	. = ..()
+	GLOB.ship_alarms += src
+
+/obj/effect/soundplayer/deltaplayer/Destroy()
 	. = ..()
-	QDEL_NULL(deltalarm)
 	GLOB.ship_alarms -= src
+	QDEL_NULL(deltaloop)
+
+/obj/effect/soundplayer/riverplayer
+	var/datum/looping_sound/river_loop/riverloop
```

Comment:
why are you making a new var for every subtype?
Just one var at the parent type is fine, and change it at the child level. Same with init, every one of them does new, so just have that at the parent level, and honestly the start part could just be a var/flag for whether it should start immediately.